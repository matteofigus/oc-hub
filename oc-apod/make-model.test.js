import makeModel from './make-model';
import image from './sample.image.json';
import video from './sample.video.json';

const scenarios = [image, video];

scenarios.forEach((scenario) => {
  test(`${scenario.media_type} to match snapshot`, () => {
    expect(makeModel(scenario)).toMatchSnapshot();
  });
});
