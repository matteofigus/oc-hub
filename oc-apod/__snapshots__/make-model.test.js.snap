// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image to match snapshot 1`] = `
Object {
  "copyright": "Alyn Wallace Photography",
  "date": "2017-03-20",
  "explanation": "Yes, but can your tree do this?  Pictured is a visual coincidence between the dark branches of a nearby tree and bright glow of a distant aurora. The beauty of the aurora -- combined with how it seemed to mimic a tree right nearby -- mesmerized the photographer to such a degree that he momentarily forgot to take pictures. When viewed at the right angle, it seemed that this tree had aurora for leaves! Fortunately, before the aurora morphed into a different overall shape, he came to his senses and capture the awe-inspiring momentary coincidence.  Typically triggered by solar explosions, aurora are caused by high energy electrons impacting the Earth's atmosphere around 150 kilometers up.  The unusual Earth-sky collaboration was witnessed earlier this month in Iceland.    Follow APOD on: Facebook,  Google Plus,  Instagram, or Twitter",
  "hdurl": "http://apod.nasa.gov/apod/image/1703/AuroraTree_Wallace_2048.jpg",
  "media_type": "image",
  "service_version": "v1",
  "style": "background-image: url(//apod.nasa.gov/apod/image/1703/AuroraTree_Wallace_960.jpg);",
  "title": "The Aurora Tree",
  "url": "//apod.nasa.gov/apod/image/1703/AuroraTree_Wallace_960.jpg",
}
`;

exports[`video to match snapshot 1`] = `
Object {
  "date": "2017-03-20",
  "explanation": "When does the line between day and night become vertical? Tomorrow. Tomorrow is an equinox on planet Earth, a time of year when day and night are most nearly equal. At an equinox, the Earth's terminator -- the dividing line between day and night -- becomes vertical and connects the north and south poles. The featured time-lapse video demonstrates this by displaying an entire year on planet Earth in twelve seconds. From geosynchronous orbit, the Meteosat satellite recorded these infrared images of the Earth every day at the same local time.  The video started at the September 2010 equinox with the terminator line being vertical. As the Earth revolved around the Sun, the terminator was seen to tilt in a way that provides less daily sunlight to the northern hemisphere, causing winter in the north. As the year progressed, the March 2011 equinox arrived halfway through the video, followed by the terminator tilting the other way, causing winter in the southern hemisphere -- and summer in the north. The captured year ends again with the September equinox, concluding another of billions of trips the Earth has taken -- and will take -- around the Sun.",
  "media_type": "video",
  "service_version": "v1",
  "style": "background-color: pink;",
  "title": "Equinox on a Spinning Earth",
  "url": "//www.youtube.com/embed/LUW51lvIFjg?rel=0&showinfo=0",
}
`;
